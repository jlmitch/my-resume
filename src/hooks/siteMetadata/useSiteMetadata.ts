import { graphql, useStaticQuery } from 'gatsby';

export const useSiteMetadata = () => {
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          title
          siteUrl
        }
      }
    }
  `);

  return {
    siteTitle: data.site.siteMetadata.title,
    siteUrl: data.site.siteMetadata.siteUrl
  }
};
