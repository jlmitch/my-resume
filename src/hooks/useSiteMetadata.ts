import React from 'react';
import { graphql, useStaticQuery } from 'gatsby';

export const useSiteMetadata = () => {
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          title
        }
      }
    }
  `);

  return {
    siteTitle: data.site.siteMetadata.title
  }
};
