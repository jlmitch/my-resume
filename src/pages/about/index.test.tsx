import React from 'react';
import { render, screen } from '@testing-library/react';
import AboutPage from '.';
import { Head } from '.';
import { useSiteMetadata } from '../../hooks/siteMetadata/useSiteMetadata';

jest.mock('../../hooks/siteMetadata/useSiteMetadata');
jest.mock('../../components/layout/Layout', () => ({children}: any) => <>{children}</>);

describe('AboutPage', () => {
  beforeEach(() => {
    (useSiteMetadata as jest.Mock).mockReturnValue({
      siteTitle: 'Michael Scott',
      siteUrl: 'https://michaelscott.com'
    });
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  test('should display AboutPage content', () => {
    const { asFragment } = render(<AboutPage />);

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByAltText('Me on my boat')).toBeInTheDocument();
  });

  test('should display Head content', () => {
    render(<Head />);

    expect(screen.getByText('About Me | Michael Scott')).toBeInTheDocument();
  });
});
