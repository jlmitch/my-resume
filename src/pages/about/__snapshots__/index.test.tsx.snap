// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage should display AboutPage content 1`] = `
<DocumentFragment>
  <img
    alt="Me on my boat"
    src="../../images/boat-me.jpg"
    style="z-index: -1; width: 250px; height: 250px; border-radius: 50%; float: left; shape-outside: circle(); margin: 0px 80px 0px 40px; box-shadow: 0 0 0vw  0.2vw #f7ffff, 0 0 0.4vw  0.2vw #a3fcf9, 0 0 2vw 0vw #178582, inset 0 0 1vw 0.2vw #178582, inset 0 0 0.2vw 0vw #1eb3ae, inset 0 0 0.3vw 0vw #f7ffff;"
  />
  <p
    class="sc-beqWaB haIifb"
  >
    As a Senior Software Engineer at KnockCRM, I bring years of experience in software development, UI engineering, and agile methodologies. Prior to my current role, I worked as a Software Engineer at Validity Inc, where I contributed to revamping an outdated product, DemandTools, by building and releasing an Electron application using Webpack, React, and Redux.
    <br />
    <br />
    Before my career in tech, I served as a Fireman in the United States Coast Guard, where I learned the importance of teamwork and effective communication.
    <br />
    <br />
    I hold a Bachelor's in Holistic Psychology from Lesley University and completed a Web Development Immersive program at General Assembly, specializing in JavaScript, Ruby, HTML, CSS, Ember.js, PostgreSQL, and MongoDB.
    <br />
    <br />
    My expertise lies in JavaScript, React, Redux, HTML and CSS, and I have experience working on agile scrum teams, leading scrum ceremonies, contributing to reusable component libraries, and providing peer-to-peer code reviews.
  </p>
</DocumentFragment>
`;
